packages/backend-data/src/convert_schema.ts

```diff
@@ -16,22 +29,63 @@ export const isModelSchema = (
   return (
     schema !== null &&
     typeof schema === 'object' &&
+    'transform' in schema &&
     typeof schema.transform === 'function'
   );
 };
 
+/**
+ * Determine if the input schema is a collection of model schemas, and perform type narrowing.
+ * @param schema the schema that might be a collection of model schemas
+ * @returns a boolean indicating whether the schema is a collection of derived model schema, with type narrowing
+ */
+export const isCombinedSchema = (
+  schema: DataSchemaInput
+): schema is DerivedCombinedSchema => {
+  return (
+    schema !== null &&
+    typeof schema === 'object' &&
+    'schemas' in schema &&
+    Array.isArray(schema.schemas)
+  );
+};
+
+/**
+ * SQL provision strategy requires a unique name per backend that is consistent between deployments
+ * @param scope construct scope
+ * @returns string provision strategy name
+ */
+const sqlProvisionStrategyName = (scope: Construct): string =>
```

Comment:
since there's nothing in here that's specific to "sql provisioning strategy", maybe this should just be called `getStableNameForDeployment` or something like that? And if we go with that, maybe we could go ahead and add that as an additional prop that gets injected [here](https://github.com/aws-amplify/amplify-backend/blob/e3a537f31e23a8dbb6e404209d7c67be9eb31627/packages/backend/src/engine/singleton_construct_container.ts#L48-L51)?

Reply:
Updated